py/picca/tests/delta_extraction/data_tests.py

```diff
@@ -126,26 +545,726 @@ def test_data_filter_forests(self):
+        self.compare_error_message(context_manager, expected_message)
 
     def test_desisim_mocks(self):
         """Test DesisimMocks"""
-        # TODO: add test
+        # case: BAO 3D
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/desi_mock_test_catalogue.fits",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 1,
+            "analysis type": "BAO 3D"
+        }})
+        for key, value in defaults_desisim_mocks.items():
+            if key not in config["data"]:
+                config["data"][key] = str(value)
+
+        data = DesisimMocks(config["data"])
+
+        self.assertTrue(len(data.forests) == 194)
+        self.assertTrue(data.min_num_pix == 50)
+        self.assertTrue(data.analysis_type == "BAO 3D")
+        self.assertTrue(
+            all(isinstance(forest, DesiForest) for forest in data.forests))
+
+        # case: Pk 1D
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/desi_mock_test_catalogue.fits",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 1,
+            "analysis type": "PK 1D"
+        }})
+        for key, value in defaults_desisim_mocks.items():
+            if key not in config["data"]:
+                config["data"][key] = str(value)
+
+        data = DesisimMocks(config["data"])
+
+        self.assertTrue(len(data.forests) == 194)
+        self.assertTrue(data.min_num_pix == 50)
+        self.assertTrue(data.analysis_type == "PK 1D")
+        self.assertTrue(
+            all(isinstance(forest, DesiPk1dForest) for forest in data.forests))
+
+    def test_desisim_mocks_read_data(self):
+        """Test method read_data from DesisimMocks"""
+        # run with one processor
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/desi_mock_test_catalogue.fits",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 1,
+            "analysis type": "BAO 3D"
+        }})
+        for key, value in defaults_desisim_mocks.items():
+            if key not in config["data"]:
+                config["data"][key] = str(value)
+
+        data = DesisimMocks(config["data"])
+
+        self.assertTrue(len(data.forests) == 194)
+
+        # run with 0 processors
+        config = ConfigParser()
+        config.read_dict({"data": {
+            "catalogue": f"{THIS_DIR}/data/desi_mock_test_catalogue.fits",
+            "keep surveys": "all",
+            "input directory": f"{THIS_DIR}/data/",
+            "out dir": f"{THIS_DIR}/results/",
+            "num processors": 0,
```

Comment:
see above

Reply:
right after this is the equivalent test with `num processors = 2`